src/ipc/ipc4/handler.c

```diff
@@ -274,7 +274,7 @@ int ipc4_pipeline_prepare(struct ipc_comp_dev *ppl_icd, uint32_t cmd)
 		switch (status) {
 		case COMP_STATE_INIT:
 			tr_dbg(&ipc_tr, "pipeline %d: reset from init", ppl_icd->id);
-			ret = ipc4_pipeline_complete(ipc, ppl_icd->id);
+			ret = ipc4_pipeline_complete(ipc, ppl_icd->id, cmd);
```

Comment:
@fkwasowi may I ask why we even need to invoke ipc4_pipeline_complete() during pipeline reset?

Reply:
This is still unanswered, but I assume this is a more complex. git blame shows this code has been here for a long, long time...

Reply:
I think this is the answer to the question: https://github.com/thesofproject/sof/commit/8541d68b507ef28b68a79713634e3b434d9f53bb